tls/s2n_client_hello.c

```diff
@@ -171,6 +198,7 @@ static int s2n_parse_client_hello(struct s2n_connection *conn)
     conn->client_protocol_version = MIN((client_protocol_version[0] * 10) + client_protocol_version[1], S2N_TLS12);
     conn->client_hello_version = conn->client_protocol_version;
 
+    GUARD(s2n_stuffer_read_uint8(in, &conn->session_id_len));
```

Comment:
You moved this line for readability?

Reply:
Yes. The protocol negotiation appearing between us reading the session_id and validation the session_id threw me for a bit when I was reading this.